cfg_mgmt/rotate.py

```diff
@@ -43,29 +42,35 @@ def delete_cfg_element(
     git_helper: gitutil.GitHelper,
     target_ref: str,
 ) -> bool:
-    did_remove = False
-    if cfg_queue_entry.target.type == 'container_registry':
+
+    delete_func: typing.Callable[[model.ConfigFactory, cmm.CfgQueueEntry], bool] = None
+
+    if (type_name := cfg_queue_entry.target.type) == 'container_registry':
         cfg_element = cfg_fac.container_registry(cfg_queue_entry.target.name)
         if cfg_element.registry_type() == om.OciRegistryType.GCR:
-            logger.info('deleting old gcr secret')
-            iam_client = ccc.gcp.create_iam_client(
-                cfg_element=cfg_element,
-            )
-            try:
-                cmg.delete_service_account_key(
-                    iam_client=iam_client,
-                    service_account_key_name=cfg_queue_entry.secretId['gcp_secret_key'],
-                )
-                did_remove = True
-            except:
-                logger.warning(f'deleting {cfg_queue_entry.secretId["gcp_secret_key"]} failed')
-                traceback.print_exc()
+            delete_func = cmg.delete_config_secret
         else:
-            logger.warning(
-                f'{cfg_element.registry_type()} is not (yet) supported for automated deletion'
-            )
-    else:
-        logger.warning(f'{cfg_queue_entry.target.type} is not (yet) supported for automated delete')
+            f'{cfg_element.registry_type()} is not (yet) supported for automated deletion'
+            return False
+
+    elif type_name == 'github':
+        delete_func = cmgh.delete_config_secret
+
+    if not delete_func:
+        logger.warning(
+            f'{type_name} is not (yet) supported for automated deletion'
+        )
+        return False
+
+    try:
+        did_remove = delete_func(cfg_factory=cfg_fac, cfg_queue_entry=cfg_queue_entry)
+    except Exception as e:
+        logger.error(
+            f"error deleting secret for cfg-type '{type_name}' with name "
+            f"'{cfg_queue_entry.target.name}': {e}."
+        )
+        traceback.print_exc()
+        did_remove = False
```

Comment:
let us rather early-exit upon errors, wdyt (thus reducing indentation level for subsequent lines of code by one)?